### PR TITLE
Resolves #809: Extract the apis and common services to a separate module

### LINKED ISSUE #809: Maintenance: versions-api
Extract api to a separate module

### PR DESCRIPTION
**Pull request by ajarmoniuk**
There's currently no clear distinction between API and service implementation, but this is the first step on the way to have it: here we're extracting a common separate module which can already act as a service layer.

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
@slawekjaranowski please review

**Comment by ajarmoniuk** [comment]
```
Caused by: org.apache.maven.wagon.providers.http.httpclient.conn.HttpHostConnectException: Connect to repo.maven.apache.org:443 [repo.maven.apache.org/146.75.28.215] failed: Connection timed out: connect
```

Could you please restart the failed jobs?

File: pom.xml [hunk 1]
```diff
@@ -99,6 +99,7 @@
     <module>versions-maven-plugin</module>
     <module>model-ruleset</module>
     <module>model-report</module>
+      <module>versions-api</module>
```
Comment (slawekjaranowski) [inline pom.xml:102]: indent?

File: versions-api/src/test/java/org/codehaus/mojo/versions/ordering/MavenVersionComparatorTest.java [hunk 2]
```diff
@@ -39,32 +40,32 @@ public MavenVersionComparatorTest()
...
     public void testSegmentCounting()
     {
-        assertThat( 3, is( instance.getSegmentCount( version( "5" ) ) ) );
-        assertThat( 3, is( instance.getSegmentCount( version( "5.0" ) ) ) );
-        assertThat( 4, is( instance.getSegmentCount( version( "5-0" ) ) ) );
-        assertThat( 1, is( instance.getSegmentCount( version( "5.3.a" ) ) ) );
-        assertThat( 1, is( instance.getSegmentCount( version( "5.0.a.1.4.5" ) ) ) );
-        assertThat( 3, is( instance.getSegmentCount( version( "" ) ) ) );
+        MatcherAssert.assertThat( 3, Is.is( instance.getSegmentCount( version( "5" ) ) ) );
+        MatcherAssert.assertThat( 3, Is.is( instance.getSegmentCount( version( "5.0" ) ) ) );
```
Comment (slawekjaranowski) [inline versions-api/src/test/java/org/codehaus/mojo/versions/ordering/MavenVersionComparatorTest.java:44]: I'm not sure if such change is needed for this PR - only make PR longer ...

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]
I still thing that code in new module it is not api ... there are many implementations.
Maybe we can start for something else ... like versons-commons - it will be more true for me - common shared code used by other module 

What do you think?

File: versions-api/src/test/java/org/codehaus/mojo/versions/ordering/MavenVersionComparatorTest.java (hunk 2 above)
Comment (ajarmoniuk) [reply versions-api/src/test/java/org/codehaus/mojo/versions/ordering/MavenVersionComparatorTest.java:44]: Ok, I'll adjust that tomorrow.

**Comment by ajarmoniuk** [comment]
Changes applied.

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**